### PR TITLE
Fixes for fuzzer bugs.

### PR DESCRIPTION
**Pull request by szabadka**
* Initialize full image area with RenderPadding if frame is outside image
* Use unaligned store/load in all stages
* Only run per-group render pipeline if modular could decode something

### CODE REVIEW

**Review by eustas — CHANGES_REQUESTED** [review]
Wait, it will cause full-image-padding for each group?

**Comment by szabadka** [comment]
> Wait, it will cause full-image-padding for each group?

Good point, I changed it to do the full padding only for group 0. PTAL

File: lib/jxl/render_pipeline/low_memory_render_pipeline.cc [hunk 1]
```diff
@@ -787,38 +787,54 @@ void LowMemoryRenderPipeline::ProcessBuffers(size_t group_id,
...
-        gx + 1 == frame_dimensions_.xsize_groups) {
-      RenderPadding(thread_id, Rect(x1, y1, full_image_xsize_ - x1,
-                                    full_image_ysize_ - y1));
+    JXL_DEBUG_V(6,
+                "Rendering padding for full image rect %s "
+                "outside group rect %s",
+                Description(full_image_rect).c_str(),
+                Description(group_rect).c_str());
+
+    if (group_id == 0 && (image_rect.xsize() == 0 || image_rect.ysize() == 0)) {
```
Comment (eustas) [inline lib/jxl/render_pipeline/low_memory_render_pipeline.cc:802]: It should be per-thread, I believe.

Comment (szabadka) [reply lib/jxl/render_pipeline/low_memory_render_pipeline.cc:802]: Whichever thread gets group-id 0, will do the initialization. This is analogous to how RenderPadding works in the other cases below. How else should I do it?

**Pull request merged**